### ISSUE TITLE
error occurred while `Lint/UnescapedBracketInRegexp` and `Style/RedundantRegexpEscape` cop with ArgumentError and IndexError since v1.80.0

### ISSUE THREAD

**Issue by r-plus** [rubocop/rubocop#14484]
## Expected behavior

can execute rubocop without error.

## Actual behavior

error occurred.

NOTE: error not occurred on rubocop v1.79.2

## Steps to reproduce the problem

test.rb
```ruby
# frozen_string_literal: true

/([^\s\(\（]+)\s*[\(\（]([^\)\）]+)[\)\）]/
```

.rubocop.yml

```ruby
AllCops:
  ParserEngine: parser_whitequark
  NewCops: enable
```

rubocop execute result

```
$ rubocop test.rb
Inspecting 1 file
An error occurred while Lint/UnescapedBracketInRegexp cop was inspecting /private/tmp/rubocop-test/test.rb:3:0.
To see the complete backtrace run rubocop -d.
An error occurred while Style/RedundantRegexpEscape cop was inspecting /private/tmp/rubocop-test/test.rb:3:0.
To see the complete backtrace run rubocop -d.
C

Offenses:

test.rb:3:7: C: [Correctable] Style/RedundantRegexpEscape: Redundant escape inside regexp literal
/([^\s\(\（]+)\s*[\(\（]([^\)\）]+)[\)\）]/
      ^^
test.rb:3:9: C: [Correctable] Style/RedundantRegexpEscape: Redundant escape inside regexp literal
/([^\s\(\（]+)\s*[\(\（]([^\)\）]+)[\)\）]/
        ^^^
test.rb:3:20: C: [Correctable] Style/RedundantRegexpEscape: Redundant escape inside regexp literal
/([^\s\(\（]+)\s*[\(\（]([^\)\）]+)[\)\）]/
                    ^^^
test.rb:3:22: C: [Correctable] Style/RedundantRegexpEscape: Redundant escape inside regexp literal
/([^\s\(\（]+)\s*[\(\（]([^\)\）]+)[\)\）]/
                       ^^
test.rb:3:30: C: [Correctable] Style/RedundantRegexpEscape: Redundant escape inside regexp literal
/([^\s\(\（]+)\s*[\(\（]([^\)\）]+)[\)\）]/
                                ^^
test.rb:3:32: C: [Correctable] Style/RedundantRegexpEscape: Redundant escape inside regexp literal
/([^\s\(\（]+)\s*[\(\（]([^\)\）]+)[\)\）]/
                                  ^^
test.rb:3:40: C: [Correctable] Style/RedundantRegexpEscape: Redundant escape inside regexp literal
/([^\s\(\（]+)\s*[\(\（]([^\)\）]+)[\)\）]/ ...


1 file inspected, 7 offenses detected, 7 offenses autocorrectable

2 errors occurred:
An error occurred while Lint/UnescapedBracketInRegexp cop was inspecting /private/tmp/rubocop-test/test.rb:3:0.
An error occurred while Style/RedundantRegexpEscape cop was inspecting /private/tmp/rubocop-test/test.rb:3:0.
Errors are usually caused by RuboCop bugs.
Please, update to the latest RuboCop version if not already in use, and report a bug if the issue still occurs on this version.
https://github.com/rubocop/rubocop/issues

Mention the following information in the issue report:
1.80.1 (using Parser 3.3.9.0, Prism 1.4.0, parser_whitequark, rubocop-ast 1.46.0, analyzing as Ruby 3.4, running on ruby 3.4.5) [arm64-darwin24]
```

<details>
<summary>with `-d` option</summary>

```
$ rubocop -d test.rb
For /private/tmp/rubocop-test: configuration from /private/tmp/rubocop-test/.rubocop.yml
Default configuration from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/config/default.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /private/tmp/rubocop-test/test.rb
An error occurred while Lint/UnescapedBracketInRegexp cop was inspecting /private/tmp/rubocop-test/test.rb:3:0.
/Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cop/lint/unescaped_bracket_in_regexp.rb:69:in 'String#scan': invalid byte sequence in UTF-8 (ArgumentError)

          expr.text.scan(/(?<!\\)\]/) do
                         ^^^^^^^^^^^
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cop/lint/unescaped_bracket_in_regexp.rb:69:in 'RuboCop::Cop::Lint::UnescapedBracketInRegexp#detect_offenses'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cop/lint/unescaped_bracket_in_regexp.rb:48:in 'block (2 levels) in RuboCop::Cop::Lint::UnescapedBracketInRegexp#on_regexp'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/regexp_parser-2.10.0/lib/regexp_parser/expression/methods/traverse.rb:75:in 'block in Regexp::Expression::Subexpression#each_expression_without_index'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/regexp_parser-2.10.0/lib/regexp_parser/expression/subexpression.rb:28:in 'Array#each'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/regexp_parser-2.10.0/lib/regexp_parser/expression/subexpression.rb:28:in 'Regexp::Expression::Subexpression#each'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/regexp_parser-2.10.0/lib/regexp_parser/expression/methods/traverse.rb:74:in 'Regexp::Expression::Subexpression#each_expression_without_index'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/regexp_parser-2.10.0/lib/regexp_parser/expression/methods/traverse.rb:76:in 'block in Regexp::Expression::Subexpression#each_expression_without_index'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/regexp_parser-2.10.0/lib/regexp_parser/expression/subexpression.rb:28:in 'Array#each'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/regexp_parser-2.10.0/lib/regexp_parser/expression/subexpression.rb:28:in 'Regexp::Expression::Subexpression#each'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/regexp_parser-2.10.0/lib/regexp_parser/expression/methods/traverse.rb:74:in 'Regexp::Expression::Subexpression#each_expression_without_index'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/regexp_parser-2.10.0/lib/regexp_parser/expression/methods/traverse.rb:76:in 'block in Regexp::Expression::Subexpression#each_expression_without_index'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/regexp_parser-2.10.0/lib/regexp_parser/expression/subexpression.rb:28:in 'Array#each'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/regexp_parser-2.10.0/lib/regexp_parser/expression/subexpression.rb:28:in 'Regexp::Expression::Subexpression#each'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/regexp_parser-2.10.0/lib/regexp_parser/expression/methods/traverse.rb:74:in 'Regexp::Expression::Subexpression#each_expression_without_index'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/regexp_parser-2.10.0/lib/regexp_parser/expression/methods/traverse.rb:13:in 'Regexp::Expression::Subexpression#each_expression'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cop/lint/unescaped_bracket_in_regexp.rb:47:in 'block in RuboCop::Cop::Lint::UnescapedBracketInRegexp#on_regexp'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/util.rb:11:in 'RuboCop::Util.silence_warnings'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cop/lint/unescaped_bracket_in_regexp.rb:46:in 'RuboCop::Cop::Lint::UnescapedBracketInRegexp#on_regexp'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cop/commissioner.rb:107:in 'Kernel#public_send'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cop/commissioner.rb:107:in 'block (2 levels) in RuboCop::Cop::Commissioner#trigger_responding_cops'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cop/commissioner.rb:171:in 'RuboCop::Cop::Commissioner#with_cop_error_handling'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cop/commissioner.rb:106:in 'block in RuboCop::Cop::Commissioner#trigger_responding_cops'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cop/commissioner.rb:105:in 'Array#each'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cop/commissioner.rb:105:in 'RuboCop::Cop::Commissioner#trigger_responding_cops'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cop/commissioner.rb:69:in 'RuboCop::Cop::Commissioner#on_regexp'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-ast-1.46.0/lib/rubocop/ast/traversal.rb:20:in 'RuboCop::AST::Traversal#walk'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cop/commissioner.rb:87:in 'RuboCop::Cop::Commissioner#investigate'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cop/team.rb:174:in 'RuboCop::Cop::Team#investigate_partial'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cop/team.rb:108:in 'RuboCop::Cop::Team#investigate'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/runner.rb:350:in 'block in RuboCop::Runner#inspect_file'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/runner.rb:349:in 'Array#each'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/runner.rb:349:in 'Enumerable#flat_map'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/runner.rb:349:in 'RuboCop::Runner#inspect_file'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/runner.rb:291:in 'block in RuboCop::Runner#do_inspection_loop'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/runner.rb:326:in 'block in RuboCop::Runner#iterate_until_no_changes'
	from <internal:kernel>:168:in 'Kernel#loop'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/runner.rb:319:in 'RuboCop::Runner#iterate_until_no_changes'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/runner.rb:287:in 'RuboCop::Runner#do_inspection_loop'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/runner.rb:167:in 'block in RuboCop::Runner#file_offenses'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/runner.rb:192:in 'RuboCop::Runner#file_offense_cache'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/runner.rb:166:in 'RuboCop::Runner#file_offenses'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/runner.rb:154:in 'RuboCop::Runner#process_file'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/runner.rb:135:in 'block in RuboCop::Runner#each_inspected_file'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/runner.rb:134:in 'Array#each'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/runner.rb:134:in 'Enumerable#reduce'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/runner.rb:134:in 'RuboCop::Runner#each_inspected_file'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/runner.rb:120:in 'RuboCop::Runner#inspect_files'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/runner.rb:73:in 'RuboCop::Runner#run'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cli/command/execute_runner.rb:26:in 'block in RuboCop::CLI::Command::ExecuteRunner#execute_runner'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cli/command/execute_runner.rb:52:in 'RuboCop::CLI::Command::ExecuteRunner#with_redirect'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cli/command/execute_runner.rb:25:in 'RuboCop::CLI::Command::ExecuteRunner#execute_runner'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cli/command/execute_runner.rb:17:in 'RuboCop::CLI::Command::ExecuteRunner#run'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cli/command.rb:11:in 'RuboCop::CLI::Command.run'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cli/environment.rb:18:in 'RuboCop::CLI::Environment#run'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cli.rb:128:in 'RuboCop::CLI#run_command'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cli.rb:135:in 'RuboCop::CLI#execute_runners'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cli.rb:54:in 'block in RuboCop::CLI#run'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cli.rb:87:in 'RuboCop::CLI#profile_if_needed'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cli.rb:45:in 'RuboCop::CLI#run'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/exe/rubocop:15:in '<top (required)>'
	from /Users/user_r0107/.rbenv/versions/3.4.5/bin/rubocop:25:in 'Kernel#load'
	from /Users/user_r0107/.rbenv/versions/3.4.5/bin/rubocop:25:in '<main>'
An error occurred while Style/RedundantRegexpEscape cop was inspecting /private/tmp/rubocop-test/test.rb:3:0.
/Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/parser-3.3.9.0/lib/parser/source/tree_rewriter.rb:406:in 'Parser::Source::TreeRewriter#check_range_validity': The range 72...73 is outside the bounds of the source (IndexError)
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cop/corrector.rb:120:in 'RuboCop::Cop::Corrector#check_range_validity'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/parser-3.3.9.0/lib/parser/source/tree_rewriter.rb:398:in 'Parser::Source::TreeRewriter#combine'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/parser-3.3.9.0/lib/parser/source/tree_rewriter.rb:194:in 'Parser::Source::TreeRewriter#replace'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/parser-3.3.9.0/lib/parser/source/tree_rewriter.rb:218:in 'Parser::Source::TreeRewriter#remove'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cop/corrector.rb:66:in 'RuboCop::Cop::Corrector#remove_leading'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cop/style/redundant_regexp_escape.rb:53:in 'block (2 levels) in RuboCop::Cop::Style::RedundantRegexpEscape#on_regexp'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cop/base.rb:426:in 'RuboCop::Cop::Base#correct'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cop/base.rb:210:in 'RuboCop::Cop::Base#add_offense'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cop/style/redundant_regexp_escape.rb:52:in 'block in RuboCop::Cop::Style::RedundantRegexpEscape#on_regexp'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cop/style/redundant_regexp_escape.rb:100:in 'block in RuboCop::Cop::Style::RedundantRegexpEscape#each_escape'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/regexp_parser-2.10.0/lib/regexp_parser/expression/methods/traverse.rb:39:in 'block in Regexp::Expression::Subexpression#traverse'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/regexp_parser-2.10.0/lib/regexp_parser/expression/subexpression.rb:28:in 'Array#each'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/regexp_parser-2.10.0/lib/regexp_parser/expression/subexpression.rb:28:in 'Regexp::Expression::Subexpression#each'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/regexp_parser-2.10.0/lib/regexp_parser/expression/methods/traverse.rb:37:in 'Enumerable#each_with_index'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/regexp_parser-2.10.0/lib/regexp_parser/expression/methods/traverse.rb:37:in 'Regexp::Expression::Subexpression#traverse'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/regexp_parser-2.10.0/lib/regexp_parser/expression/methods/traverse.rb:42:in 'block in Regexp::Expression::Subexpression#traverse'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/regexp_parser-2.10.0/lib/regexp_parser/expression/subexpression.rb:28:in 'Array#each'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/regexp_parser-2.10.0/lib/regexp_parser/expression/subexpression.rb:28:in 'Regexp::Expression::Subexpression#each'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/regexp_parser-2.10.0/lib/regexp_parser/expression/methods/traverse.rb:37:in 'Enumerable#each_with_index'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/regexp_parser-2.10.0/lib/regexp_parser/expression/methods/traverse.rb:37:in 'Regexp::Expression::Subexpression#traverse'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cop/style/redundant_regexp_escape.rb:99:in 'Enumerator#each'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cop/style/redundant_regexp_escape.rb:99:in 'Enumerable#reduce'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cop/style/redundant_regexp_escape.rb:99:in 'RuboCop::Cop::Style::RedundantRegexpEscape#each_escape'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cop/style/redundant_regexp_escape.rb:46:in 'RuboCop::Cop::Style::RedundantRegexpEscape#on_regexp'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cop/commissioner.rb:107:in 'Kernel#public_send'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cop/commissioner.rb:107:in 'block (2 levels) in RuboCop::Cop::Commissioner#trigger_responding_cops'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cop/commissioner.rb:171:in 'RuboCop::Cop::Commissioner#with_cop_error_handling'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cop/commissioner.rb:106:in 'block in RuboCop::Cop::Commissioner#trigger_responding_cops'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cop/commissioner.rb:105:in 'Array#each'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cop/commissioner.rb:105:in 'RuboCop::Cop::Commissioner#trigger_responding_cops'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cop/commissioner.rb:69:in 'RuboCop::Cop::Commissioner#on_regexp'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-ast-1.46.0/lib/rubocop/ast/traversal.rb:20:in 'RuboCop::AST::Traversal#walk'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cop/commissioner.rb:87:in 'RuboCop::Cop::Commissioner#investigate'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cop/team.rb:174:in 'RuboCop::Cop::Team#investigate_partial'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cop/team.rb:108:in 'RuboCop::Cop::Team#investigate'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/runner.rb:350:in 'block in RuboCop::Runner#inspect_file'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/runner.rb:349:in 'Array#each'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/runner.rb:349:in 'Enumerable#flat_map'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/runner.rb:349:in 'RuboCop::Runner#inspect_file'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/runner.rb:291:in 'block in RuboCop::Runner#do_inspection_loop'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/runner.rb:326:in 'block in RuboCop::Runner#iterate_until_no_changes'
	from <internal:kernel>:168:in 'Kernel#loop'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/runner.rb:319:in 'RuboCop::Runner#iterate_until_no_changes'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/runner.rb:287:in 'RuboCop::Runner#do_inspection_loop'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/runner.rb:167:in 'block in RuboCop::Runner#file_offenses'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/runner.rb:192:in 'RuboCop::Runner#file_offense_cache'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/runner.rb:166:in 'RuboCop::Runner#file_offenses'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/runner.rb:154:in 'RuboCop::Runner#process_file'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/runner.rb:135:in 'block in RuboCop::Runner#each_inspected_file'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/runner.rb:134:in 'Array#each'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/runner.rb:134:in 'Enumerable#reduce'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/runner.rb:134:in 'RuboCop::Runner#each_inspected_file'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/runner.rb:120:in 'RuboCop::Runner#inspect_files'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/runner.rb:73:in 'RuboCop::Runner#run'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cli/command/execute_runner.rb:26:in 'block in RuboCop::CLI::Command::ExecuteRunner#execute_runner'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cli/command/execute_runner.rb:52:in 'RuboCop::CLI::Command::ExecuteRunner#with_redirect'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cli/command/execute_runner.rb:25:in 'RuboCop::CLI::Command::ExecuteRunner#execute_runner'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cli/command/execute_runner.rb:17:in 'RuboCop::CLI::Command::ExecuteRunner#run'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cli/command.rb:11:in 'RuboCop::CLI::Command.run'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cli/environment.rb:18:in 'RuboCop::CLI::Environment#run'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cli.rb:128:in 'RuboCop::CLI#run_command'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cli.rb:135:in 'RuboCop::CLI#execute_runners'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cli.rb:54:in 'block in RuboCop::CLI#run'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cli.rb:87:in 'RuboCop::CLI#profile_if_needed'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/lib/rubocop/cli.rb:45:in 'RuboCop::CLI#run'
	from /Users/user_r0107/.rbenv/versions/3.4.5/lib/ruby/gems/3.4.0/gems/rubocop-1.80.1/exe/rubocop:15:in '<top (required)>'
	from /Users/user_r0107/.rbenv/versions/3.4.5/bin/rubocop:25:in 'Kernel#load'
	from /Users/user_r0107/.rbenv/versions/3.4.5/bin/rubocop:25:in '<main>'
C

Offenses:

test.rb:3:7: C: [Correctable] Style/RedundantRegexpEscape: Redundant escape inside regexp literal
/([^\s\(\（]+)\s*[\(\（]([^\)\）]+)[\)\）]/
      ^^
test.rb:3:9: C: [Correctable] Style/RedundantRegexpEscape: Redundant escape inside regexp literal
/([^\s\(\（]+)\s*[\(\（]([^\)\）]+)[\)\）]/
        ^^^
test.rb:3:20: C: [Correctable] Style/RedundantRegexpEscape: Redundant escape inside regexp literal
/([^\s\(\（]+)\s*[\(\（]([^\)\）]+)[\)\）]/
                    ^^^
test.rb:3:22: C: [Correctable] Style/RedundantRegexpEscape: Redundant escape inside regexp literal
/([^\s\(\（]+)\s*[\(\（]([^\)\）]+)[\)\）]/
                       ^^
test.rb:3:30: C: [Correctable] Style/RedundantRegexpEscape: Redundant escape inside regexp literal
/([^\s\(\（]+)\s*[\(\（]([^\)\）]+)[\)\）]/
                                ^^
test.rb:3:32: C: [Correctable] Style/RedundantRegexpEscape: Redundant escape inside regexp literal
/([^\s\(\（]+)\s*[\(\（]([^\)\）]+)[\)\）]/
                                  ^^
test.rb:3:40: C: [Correctable] Style/RedundantRegexpEscape: Redundant escape inside regexp literal
/([^\s\(\（]+)\s*[\(\（]([^\)\）]+)[\)\）]/ ...


1 file inspected, 7 offenses detected, 7 offenses autocorrectable

2 errors occurred:
An error occurred while Lint/UnescapedBracketInRegexp cop was inspecting /private/tmp/rubocop-test/test.rb:3:0.
An error occurred while Style/RedundantRegexpEscape cop was inspecting /private/tmp/rubocop-test/test.rb:3:0.
configuration from /private/tmp/rubocop-test/.rubocop.yml
Errors are usually caused by RuboCop bugs.
Please, update to the latest RuboCop version if not already in use, and report a bug if the issue still occurs on this version.
https://github.com/rubocop/rubocop/issues

Mention the following information in the issue report:
1.80.1 (using Parser 3.3.9.0, Prism 1.4.0, parser_whitequark, rubocop-ast 1.46.0, analyzing as Ruby 3.4, running on ruby 3.4.5) [arm64-darwin24]
Finished in 0.07388000001083128 seconds

```

</details>

## RuboCop version

```
 rubocop -V
1.80.1 (using Parser 3.3.9.0, Prism 1.4.0, parser_whitequark, rubocop-ast 1.46.0, analyzing as Ruby 3.4, running on ruby 3.4.5) [arm64-darwin24]
```

**Comment by Earlopain** [comment]
Thanks for the report. I believe this is a bug in the library we use to parse regexp. https://github.com/ammar/regexp_parser/issues/104

The reason why it started showing up only in 1.80.0 is because there was a bug that resulted in the cop not running in the first place: https://github.com/rubocop/rubocop/issues/14435

**Comment by Earlopain** [comment]
This is fixed in `regexp_parser` v2.11.3

**Comment by r-plus** [comment]
thanks, error is gone when update the `regex_parser`.
